pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -168,3 +159,58 @@ func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidat
 
 	return deletedLS, async, err
 }
+
+// Update overrides the generic behavior of the Storage and writes first to the legacy storage and then to US.
+// Update overrides the behavior of the generic DualWriter and writes first to LegacyStorage and then to Storage.
+func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.Updater)
+	if !ok {
+		return nil, false, errDualWriterUpdaterMissing
+	}
+
+	// get old and new (updated) object so they can be stored in legacy store
+	old, err := d.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.FromContext(ctx).Error(err, "could not get object", "mode", Mode2)
+			return nil, false, err
+		}
+		klog.FromContext(ctx).Error(err, "object not found for update, creating one", "mode", Mode2)
+		forceAllowCreate = true
+	}
+
+	updated, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return nil, false, err
+	}
+
+	obj, created, err := legacy.Update(ctx, name, &updateWrapper{upstream: objInfo, updated: updated}, createValidation, updateValidation, forceAllowCreate, options)
```

Comment:
Do we not need to clear the RV and UID before the call to legacy.Update? I think I had tried it out without clearing, but I don't remember the outcome. These comments are giving me pause though: https://github.com/grafana/grafana/blob/main/pkg/apiserver/rest/dualwriter.go#L149-L150

By the way, the way I tried it out manually is described under "reproduction steps" in https://github.com/grafana/grafana/pull/84605

Reply:
from what I understood we have to, because the `resource_version` might have changed between obj.Storage.Get() and obj.Storage.Update(). But I think we need to ensure the both resource_version and UID are the same between the legacyStorage and the Storage, otherwise the mapping for the same object between the 2 stores would be lost.
There's a conversation about that in [here](https://raintank-corp.slack.com/archives/C06JFJ7H3GC/p1712742127308989).
In any case, I think it's one of those things we can investigate later and get rid of if we see it's not needed :)